### PR TITLE
Signon: Enable new-style logging

### PR DESCRIPTION
**Pull request by boffbowsh**
This updates the log collection to collect the application logs from
stdout and stderr instead of reading from logfiles that the app creates
itself.

To be deployed after https://github.com/alphagov/signonotron2/pull/464

### CODE REVIEW

**Comment by surminus** [comment]
:+1: merge when green

File: modules/govuk/manifests/apps/signon.pp [hunk 1]
```diff
@@ -37,7 +37,7 @@
     port               => $port,
     vhost_ssl_only     => true,
     health_check_path  => '/users/sign_in',
-    log_format_is_json => true,
+    legacy_logging     => false,
     vhost_aliases      => ['signonotron'],
     logstream          => $ensure_logstream,
```
Comment (alexmuller) [inline modules/govuk/manifests/apps/signon.pp]: For reference, this param (set in hieradata) was [the one @bradwright was referring to](https://github.com/alphagov/signonotron2/pull/464#issuecomment-177001002). Do we want to set it to true?

**Comment by alexmuller** [comment]
There's lint on this pull request which I think is just the fat arrows not lining up.

File: modules/govuk/manifests/apps/signon.pp (hunk 1 above)
Comment (bradwright) [inline modules/govuk/manifests/apps/signon.pp]: I think we do.

**Comment by boffbowsh** [comment]
Fixed

File: modules/govuk/manifests/apps/signon.pp (hunk 1 above)
Comment (boffbowsh) [inline modules/govuk/manifests/apps/signon.pp]: Should we explicitly set that to true here, or just in hierdata/

Comment (alexmuller) [inline modules/govuk/manifests/apps/signon.pp]: The `logstream` param on `govuk::app` defaults to true so I think we can delete the param from this class and from hieradata entirely.

Comment (boffbowsh) [inline modules/govuk/manifests/apps/signon.pp]: Done (will remove from hieradata when this is deployed)

**Comment by alexmuller** [comment]
The change looks good to me but the commit to fix arrow alignment also does a whole lot more (sorry to nitpick commit history)

**Comment by boffbowsh** [comment]
@alexmuller fixed I am bad at rebasing

**Comment by alexmuller** [comment]
Looks like Puppet to me, merge at will :+1:

**Pull request merged**